pkg/serving/config_changes_test.go

```diff
@@ -139,6 +139,43 @@ func testUpdateEnvVarsModify(t *testing.T, revision *servingv1alpha1.RevisionTem
 	}
 }
 
+func TestUpdateContainerPort(t *testing.T) {
+	template, _ := getV1alpha1Config()
+	err := UpdateContainerPort(template, 8080)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Verify update is successful or not
+	for p, _ := range template.Spec.Containers {
+		for q, _ := range template.Spec.Containers[p].Ports {
+			if template.Spec.Containers[p].Ports[q].ContainerPort != 8080 {
+				t.Error("Failed to update the container port")
+			}
+		}
+	}
+
+	// update template with container port info
+	for i, _ := range template.Spec.Containers {
+		for j, _ := range template.Spec.Containers[i].Ports {
+			template.Spec.Containers[i].Ports[j].ContainerPort = 9090
+		}
+	}
+
+	err = UpdateContainerPort(template, 80)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that given port overrides the existing container port
+	for p, _ := range template.Spec.Containers {
```

Comment:
assert again, that there is only 1 container and 1 port after the update (you could put that into an extrac func as its needed twice)